aquarius/events/events_monitor.py

```diff
@@ -574,70 +505,105 @@ def get_assets_in_chain(self):
-
-        if event_name == EventTypes.EVENT_METADATA_CREATED:
-            hash_text = "MetadataCreated(address,uint8,string,bytes,bytes,bytes32,uint256,uint256)"
-        elif event_name == EventTypes.EVENT_METADATA_UPDATED:
-            hash_text = "MetadataUpdated(address,uint8,string,bytes,bytes,bytes32,uint256,uint256)"
-        elif event_name == EventTypes.EVENT_METADATA_STATE:
-            hash_text = "MetadataState(address,uint8,uint256,uint256)"
-        elif event_name == EventTypes.EVENT_TOKEN_URI_UPDATE:
-            hash_text = "TokenURIUpdate(address,string,uint256,uint256,uint256)"
-        elif event_name == EventTypes.EVENT_EXCHANGE_CREATED:
-            hash_text = "ExchangeCreated(bytes32,address,address,address,uint256)"
-        elif event_name == EventTypes.EVENT_EXCHANGE_RATE_CHANGED:
-            hash_text = "ExchangeRateChanged(bytes32,address,uint256)"
-        elif event_name == EventTypes.EVENT_DISPENSER_CREATED:
-            hash_text = "DispenserCreated(address,address,uint256,uint256,address)"
-        elif event_name == EventTypes.EVENT_TRANSFER:
-            hash_text = "Transfer(address,address,uint256)"
-        else:
-            hash_text = (
-                "OrderStarted(address,address,uint256,uint256,uint256,address,uint256)"
-            )
-
-        event_signature_hash = self._web3.keccak(text=hash_text).hex()
-
-        _from = from_block
-        _to = min(_from + chunk_size - 1, to_block)
-
-        logger.debug(
-            f"Searching for {event_name} events on chain {self._chain_id} "
+    def get_event_logs(self, from_block, to_block):
+        """Get all events from -> to and process them.
+        If that fails, and we tried with multiple blocks, let split handle it
+        If that fails, and we tried on a single block, then try to get events one by one instead of all
+        """
+        logger.info(
+            f"Searching for events events on chain {self._chain_id} "
             f"in blocks {from_block} to {to_block}."
         )
 
         filter_params = {
-            "topics": [event_signature_hash],
-            "fromBlock": _from,
-            "toBlock": _to,
+            "topics": [list(EventTypes.hashes.keys())],
+            "fromBlock": from_block,
+            "toBlock": to_block,
         }
 
-        all_logs = []
-        while _from <= to_block:
-            # Search current chunk
+        try:
             logs = self._web3.eth.get_logs(filter_params)
-            all_logs.extend(logs)
-            if (_from - from_block) % 1000 == 0:
-                logger.debug(
-                    f"Searched blocks {_from} to {_to} on chain {self._chain_id}; "
-                    f"{len(all_logs)} {event_name} events detected so far."
-                )
+        except Exception as e:
+            if from_block < to_block:
+                # splitting in two might help, so rely on that
+                raise Exception(f"Failed to get events for multiple blocks")
```

Comment:
Not sure if needed, because there is only one piece of code that handles that (Line 242)
```
       try:
            self.get_event_logs(from_block, to_block)
        except Exception as e:
            logger.info(f"Failed to get events from {from_block} to {to_block}")
```